crates/core/tedge_api/src/entity_store.rs

```diff
@@ -501,6 +503,39 @@ impl TryFrom<&Message> for EntityRegistrationMessage {
     }
 }
 
+impl From<&EntityRegistrationMessage> for Message {
+    fn from(value: &EntityRegistrationMessage) -> Self {
+        let entity_topic_id = value.topic_id.clone();
+
+        let mut register_payload: Map<String, Value> = Map::new();
+
+        let entity_type = match value.r#type {
+            EntityType::MainDevice => "device",
+            EntityType::ChildDevice => "child-device",
+            EntityType::Service => "service",
+        };
+        register_payload.insert("@type".into(), Value::String(entity_type.to_string()));
+
+        if let Some(external_id) = &value.external_id {
+            register_payload.insert("@id".into(), Value::String(external_id.as_ref().into()));
+        }
+
+        if let Some(parent_id) = &value.parent {
+            register_payload.insert("@parent".into(), Value::String(parent_id.to_string()));
+        }
+
+        if let Value::Object(other_keys) = value.payload.clone() {
+            register_payload.extend(other_keys)
+        }
+
+        Message::new(
+            &Topic::new(&format!("{MQTT_ROOT}/{entity_topic_id}")).unwrap(),
```

Comment:
The `MQTT_ROOT` should not be hard-coded.

Reply:
Moved the this function to the converter where the schema is available.